tests/unit/test_parser.py

```diff
@@ -16,8 +16,10 @@
 )
 def test_parsed_data_item(name, raw_value, unit, derived_value, short_description, long_description, valid):
     """Test ParsedDataItem"""
-    if valid:
-        pdi = parser.ParsedDataItem(name, raw_value, unit, derived_value, short_description, long_description)
-    else:
-        with pytest.raises(ValueError):
-            pdi = parser.ParsedDataItem(name, raw_value, unit, derived_value, short_description, long_description)
```

Comment:
Note that the final commit did remove this ValueError being raised from the class creation.

I profiled the code and the `__postinit__` was being called for every single creation of the items (added about 0.2 seconds to a 6 second run), so it is in the hot loop and just checking whether the name or raw value is None. IMO, this isn't needed and if we create a data item with None in either of those that is OK and just what we got from the parser.